### PR TITLE
support for D7 and persistent maps/no go lines

### PR DESCRIPTION
**Pull request by nathanfaber**
Same initial point as @dshokouhi #14 to support basic-3 but also support category 4 for persistent and no-go lines.

### CODE REVIEW

**Comment by dshokouhi** [comment]
Excellent! I was looking at this earlier today :) I also realized that we apparently have a "deep" cleaning navigation mode?  I was going to check that out and see how much better it actually cleans since I do not see the option in the app (https://developers.neatorobotics.com/api/robot-remote-protocol/housecleaning) Ideally I think we should let the user configure this and maybe even call the proper mode upon running it.  Personally I don't like extra care so I switched to normal on my install.

File: pybotvac/robot.py [hunk 1]
```diff
@@ -56,11 +56,16 @@ def _message(self, json):
...
+    def start_cleaning(self, mode=2, navigation_mode=2, category=None):
         # mode & naivigation_mode used if applicable to service version
         # mode: 1 eco, 2 turbo
         # navigation_mode: 1 normal, 2 extra care
-        
+        # category: 2 non-persistent map, 4 persistent map
+
+        #Default to using the persistent map if we support basic-3.
+        if category is None:
+            category = 3 if self.service_version == 'basic-3' else 2
```
Comment (dshokouhi) [inline pybotvac/robot.py]: it should be `category = 4` right?  `3` is for spot cleaning

Comment (dshokouhi) [inline pybotvac/robot.py]: I think this should be a `4` instead of a `3` because `3` is used for spot cleaning only.

Comment (nathanfaber) [reply pybotvac/robot.py]: Good catch, that should be a 4.

**Comment by nathanfaber** [comment]
Also note that for Home Assistant to like these new modes ACTIONS need to be extended for 11-15 and probably need to extend the ERRORS and ALERTS in the neato component.
https://developers.neatorobotics.com/api/robot-remote-protocol/request-response-formats
https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/neato.py

**Comment by nathanfaber** [comment]
Closing for #14

**Pull request closed without merging**